addons/mass_mailing/models/mail_mail.py

```diff
@@ -51,7 +51,13 @@ def _get_unsubscribe_url(self, cr, uid, mail, email_to, context=None):
         url = urlparse.urljoin(
             base_url, 'mail/mailing/%(mailing_id)s/unsubscribe?%(params)s' % {
                 'mailing_id': mail.mailing_id.id,
-                'params': werkzeug.url_encode({'db': cr.dbname, 'res_id': mail.res_id, 'email': email_to})
+                'params': werkzeug.url_encode({
+                    'db': cr.dbname,
+                    'res_id': mail.res_id,
+                    'email': email_to,
+                    'token': self.pool['mail.mass_mailing']._unsubscribe_token(
+                        cr, uid, mail.mailing_id.id, mail.res_id, email_to),
```

Comment:
I'm afraid you are wrong this time. The api takes care of handling calls with int or list in ids argument. Anyway, [it is supposed that old api will disappear in v10](https://www.odoo.com/es_ES/groups/community-59/community-18270712?mode=&date_begin=&date_end=).